lib/prelude.dx

```diff
@@ -857,11 +855,60 @@ splitV : Iso a ({|} | a) =
 def sliceFields (iso: Iso ({|} | a) (b | c)) (tab: a=>v) : b=>v =
   reindex (buildWith $ splitV &>> iso) tab
 
+'Dynamic buffer
+
+-- TODO: should we be able to use `Ref World Int` instead of `Ptr Int`?
+-- TODO: would be nice to be able to use records here
+data DynBuffer a:Type = MkDynBuffer (Ptr (Int & Int & Ptr a))  -- size, max size, buf ptr
```

Comment:
If that's the only place where we use the storable instance for pairs, then why can't we just make it hold three pointers?